### PR TITLE
unit test for embedded CDI SessionScoped (#1940)

### PR DESCRIPTION
**Pull request by fcorneli**

### CODE REVIEW

**Comment by janbartel** [comment]
@fcorneli  did you sign the pull request with the correct identity? Looks like you don't have a valid eclipse contribution agreement for the identity you used. Click the "Details" link on the failed check in this pull request for details.

**Comment by fcorneli** [comment]
@janbartel Signed this ECA.

File: jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/MySessionServlet.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+public class MySessionServlet extends HttpServlet
+{
+
+	@Inject
+	private MySessionClass mySessionObject;
+
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException
+	{
+		this.mySessionObject.invoke();
```
Comment (joakime) [inline jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/MySessionServlet.java]: Are you relying on an error to indicate test failure here? (the NPE)

File: jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/WeldInitializationTest.java [hunk 2]
```diff
@@ -118,4 +119,11 @@ public void testRequestParamServletAbc() throws Exception
             assertThat("Response", resp, containsString(" param[abc] = [123]"));
         }
     }
+
+    @Test
+    public void testSessionScoped() throws Exception
+    {
+        HttpURLConnection http = (HttpURLConnection) serverHttpURI.resolve("session").toURL().openConnection();
+        assertThat("response code", http.getResponseCode(), is(200));
```
Comment (joakime) [inline jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/WeldInitializationTest.java]: It seems odd to just check the status code here.  Is it really just pass/fail, no half-measure?
Does the response at least contain the stacktrace when it does fail? (would help with troubleshooting)

File: jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/MySessionServlet.java (hunk 1 above)
Comment (fcorneli) [reply jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/MySessionServlet.java]: Yes, without the patch, this unit test fails.

File: jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/WeldInitializationTest.java (hunk 2 above)
Comment (fcorneli) [reply jetty-cdi/cdi-servlet/src/test/java/org/eclipse/jetty/cdi/servlet/WeldInitializationTest.java]: A unit test that tests the session scope "persistence" would require an HTTP client that keeps track of session cookies and such. Hence I've kept it simple here, just to capture the NPE.

**Comment by janbartel** [comment]
@fcorneli  still seems to think you didn't sign the commit with the correct identity?

**Comment by joakime** [comment]
Closing due to lack reply about git signoff (to satisfy [Eclipse DCO](https://www.eclipse.org/legal/DCO.php) requirements)

See https://wiki.eclipse.org/Development_Resources/Contributing_via_Git

**Pull request closed without merging**